protocol/daemons/liquidation/client/client.go

```diff
@@ -65,262 +84,55 @@ func Start(
-) (
-	subaccounts []satypes.Subaccount,
-	err error,
+	s SubTaskRunner,
+	flags flags.DaemonFlags,
+	ticker *time.Ticker,
+	done <-chan bool,
+	subaccountQueryClient satypes.QueryClient,
+	clobQueryClient clobtypes.QueryClient,
+	liquidationServiceClient api.LiquidationServiceClient,
 ) {
-	defer telemetry.ModuleMeasureSince(metrics.LiquidationDaemon, time.Now(), metrics.GetAllSubaccounts, metrics.Latency)
-	subaccounts = make([]satypes.Subaccount, 0)
-
-	var nextKey []byte
 	for {
-		subaccountsFromKey, next, err := getSubaccountsFromKey(
-			ctx,
-			client,
-			limit,
-			nextKey,
-		)
-
-		if err != nil {
-			return nil, err
-		}
-
-		subaccounts = append(subaccounts, subaccountsFromKey...)
-		nextKey = next
-
-		if len(nextKey) == 0 {
-			break
-		}
-	}
-
-	telemetry.ModuleSetGauge(
-		metrics.LiquidationDaemon,
-		float32(len(subaccounts)),
-		metrics.GetAllSubaccounts,
-		metrics.Count,
-	)
-
-	return subaccounts, nil
-}
-
-// GetLiquidatableSubaccountIds verifies collateralization statuses of subaccounts with
-// at least one open position and returns a list of unique and potentially liquidatable subaccount ids.
-func GetLiquidatableSubaccountIds(
-	ctx context.Context,
-	client clobtypes.QueryClient,
-	liqFlags flags.LiquidationFlags,
-	subaccounts []satypes.Subaccount,
-) (
-	liquidatableSubaccountIds []satypes.SubaccountId,
-	err error,
-) {
-	defer telemetry.ModuleMeasureSince(
-		metrics.LiquidationDaemon,
-		time.Now(),
-		metrics.GetLiquidatableSubaccountIds,
-		metrics.Latency,
-	)
-
-	// Filter out subaccounts with no open positions.
-	subaccountsToCheck := make([]satypes.SubaccountId, 0)
-	for _, subaccount := range subaccounts {
-		if len(subaccount.PerpetualPositions) > 0 {
-			subaccountsToCheck = append(subaccountsToCheck, *subaccount.Id)
-		}
-	}
-
-	telemetry.ModuleSetGauge(
-		metrics.LiquidationDaemon,
-		float32(len(subaccountsToCheck)),
-		metrics.SubaccountsWithOpenPositions,
-		metrics.Count,
-	)
-
-	// Query the gRPC server in chunks of size `liqFlags.RequestChunkSize`.
-	liquidatableSubaccountIds = make([]satypes.SubaccountId, 0)
-	for start := 0; start < len(subaccountsToCheck); start += int(liqFlags.RequestChunkSize) {
-		end := lib.Min(start+int(liqFlags.RequestChunkSize), len(subaccountsToCheck))
-
-		results, err := CheckCollateralizationForSubaccounts(
-			ctx,
-			client,
-			subaccountsToCheck[start:end],
-		)
-		if err != nil {
-			return nil, err
-		}
-
-		for _, result := range results {
-			if result.IsLiquidatable {
-				liquidatableSubaccountIds = append(liquidatableSubaccountIds, result.SubaccountId)
+		select {
+		case <-ticker.C:
+			if err := s.RunLiquidationDaemonTaskLoop(
+				client,
+				ctx,
+				flags.Liquidation,
+				subaccountQueryClient,
+				clobQueryClient,
+				liquidationServiceClient,
+			); err != nil {
+				// TODO(DEC-947): Move daemon shutdown to application.
+				client.logger.Error("Liquidations daemon returned error", "error", err)
+				client.ReportFailure(err)
```

Comment:
Failure report here. This should be called as soon as an error is returned, which should happen when any gRPC request in the daemon's task loop fails - so there should be little latency between a request failure and this report.